doc/headset.md

```diff
@@ -0,0 +1,430 @@
+
+Declaration:
+``` ts
+    answerIncomingCall(conversationId: string): Promise<void>;
+```
+
+Params:
+* `conversationId: string` - The conversationId of the call that needs to be answered
+
+Returns: a Promise containing `void`
+
+#### `setMute()`
+Function that calls the headset library's `setMute(isMuted)` function. This signals the headset device to switch on or off (depending on the value of `isMuted`) the mute call button's light to show the call has been muted or unmuted respectively
+
+Declaration:
+``` ts
+    setMute(isMuted: boolean): Promise<void>;
+```
+
+Params:
+* `isMuted: boolean` - value to determine if the device is now muted or unmuted
+
+Returns: a Promise containing `void`
+
+#### `setHold()`
+Function that calls the headset library's `setHold(conversationId, isHeld)` function. This signals the headset device to switch on or off (depending the value of `isHeld`) the hold call button's light to show the call has been held or resumed respectively
+
+Declaration:
+``` ts
+    setHold(conversationId: string, isHeld: boolean): Promise<void>;
+```
+
+Params:
+* `conversationId: string` - The conversationId of the call that needs to be held or resumed
+* `isHeld: boolean` - value to determine if the device is now on hold or resumed
+
+Returns: a Promise containing `void`
+
+--------
+
+### Events
+The SDK Headset Utility does not explicitly emit events itself. It uses RxJS observables (https://rxjs.dev/guide/observable) to emit the events which are then subscribed to within the consuming app. It listens for changes and fires functions that correspond to the events
+
+### `deviceAnsweredCall`
+Event emitted when a user presses the answer call button during an incoming call on their selected device. The event includes the event `name` as it is interpretted by the headset and a collection of items that may help with logging (`event`). It can also potentially have a `code` that corresponds to the event.
+
+Declaration:
+``` ts
+    sdk.headset.headsetEvents.subscribe(event: {
+        event: 'deviceAnsweredCall',
+        payload: {
+            name: string,
+            code?: string,
+            event: { `containing various items mostly for logging purposes` }
+        }
+    } => {
+        if (event.event === 'deviceAnsweredCall') {
+            sdk.acceptPendingSession();
+        }
+    })
+```
+
+Value of event:
+* `event: EmittedHeadsetEvents` - string value emitted by the headset library to determine what event had just occurred
+* `payload:` {
+    `name: string` - Name of the recent event as interpretted by the headset device
+    `event`: { containing various items mostly for logging purposes}
+    `code?: string` - Optional: A string value of a number that represents the action that was just taken. Not all vendors supply a code which is why it is only optional
+}
```

Comment:
Not sure what styling you were aiming for here, but it just comes out as a string. That isn't bad and there is no needed to change it. Just wanted to bring it up in case you were hoping for indented new lines or something.

![image](https://user-images.githubusercontent.com/32391370/157102309-7a69781b-266d-419c-9988-a0cce0f309c1.png)

Reply:
Looks like it is that way for all the event payloads.

Reply:
I was hoping for indented... But this is still pretty readable right? Or does it need to be touched up?